File: backend/schemas/offboarding.py
"""
Schemas Pydantic para offboarding — instancias y activos.
"""
from datetime import date
from typing import List, Literal, Optional
from uuid import UUID

from pydantic import BaseModel

MotivoEgreso = Literal[
    "renuncia", "despido", "acuerdo_mutuo",
    "fin_contrato", "jubilacion", "fallecimiento", "otro",
]


class OffboardingCreate(BaseModel):
    empleado_id: UUID
    motivo: MotivoEgreso
    fecha_ultimo_dia: Optional[date] = None
    descripcion_motivo: Optional[str] = None


class ActivoUpdate(BaseModel):
    devuelto: bool


class ActivoResponse(BaseModel):
    id: UUID
    tipo_activo: str
    descripcion: Optional[str] = None
    estado: str
    devuelto: bool


class AccesoResponse(BaseModel):
    id: UUID
    tipo: str
    descripcion: Optional[str] = None
    revocado: bool


class OffboardingResponse(BaseModel):
    id: UUID
    empleado_id: UUID
    empresa_id: Optional[UUID] = None
    empresa_nombre: Optional[str] = None
    empleado_nombre: str
    motivo: str
    estado: str
    fecha_inicio: str
    progreso: int
    activos: List[ActivoResponse] = []
    accesos: List[AccesoResponse] = []

File: frontend/types/offboarding.ts
export type MotivoEgreso =
  | "renuncia"
  | "despido"
  | "acuerdo_mutuo"
  | "fin_contrato"
  | "jubilacion"
  | "fallecimiento"
  | "otro"

export interface ActivoResponse {
  id: string
  tipo_activo: string
  descripcion: string | null
  estado: string
  devuelto: boolean
}

export interface AccesoResponse {
  id: string
  tipo: string
  descripcion: string | null
  revocado: boolean
}

export interface OffboardingInstancia {
  id: string
  empleado_id: string
  empresa_id: string | null
  empresa_nombre: string | null
  empleado_nombre: string
  motivo: MotivoEgreso
  estado: string
  fecha_inicio: string
  progreso: number
  activos: ActivoResponse[]
  accesos: AccesoResponse[]
}

export interface OffboardingCreate {
  empleado_id: string
  motivo: MotivoEgreso
  fecha_ultimo_dia?: string
  descripcion_motivo?: string
}
